Align the boolean validity claim (#87)
peppelinux/draft-demarco-oauth-status-assertions: draft-demarco-oauth-status-assertions.md

    "exp": 1504785536,
    "credential_hash": $hash-about-Issuer-Signed-JWT,
    "credential_hash_alg": "sha-256",

>     "credential_status_validity": valid,

Reviewer: ```suggestion
    "credential_status_validity": "valid",
```

| **exp** | UNIX Timestamp with the expiration time of the JWT. It MUST be greater than the value set for `iat`. | {{RFC9126}}, {{RFC7519}}, {{RFC7515}} |
| **credential_hash** | Hash value of the Digital Credential the Status Assertion is bound to. | this specification |
| **credential_hash_alg** | The Algorithm used of hashing the Digital Credential to which the Status Assertion is bound. The value SHOULD be set to `sha-256`. | this specification |

> | **credential_status_validity**| Numerical value indicating the validity of the Credential linked to the Status Assertion, describing its state, mode, condition or stage. The value MUST be from the IANA registry (TBD). Status validity parameter is REQUIRED, and the Verifier MUST verify its presence and value to assess the Credential's validity. | this specification |

Reviewer 2: ```suggestion
| **credential_status_validity**| Numerical value indicating the validity of the Credential linked to the Status Assertion, describing its state, mode, condition or stage. The value MUST be from the IANA registry (TBD).Values from 0 to 2 are defined by this specification with the following meaning: 0-VALID, 1-INVALID, 2-SUSPENDED. Credential Issuers MAY extend with other numeric values based on use cases of interest. Status validity parameter is REQUIRED, and the Verifier MUST verify its presence and value to assess the Credential's validity. | this specification |
```

Author: I think that we agreed that we don't want to hardcode the values in this spec, as we want to align values with related parameter in Status List specification. We assume that they will register the values in the IANA registry so we want to reference to that instead. I think thought that we can use this values in the examples.

Reviewer 2: Even if we will have the values registered in IANA, why shouldn't we have in this specification? IMHO having them explicitly here could be useful for implementers that are already implementing this. Am I missing something?

Reviewer: I created this issue, https://github.com/peppelinux/draft-demarco-oauth-status-assertions/issues/90

and am resolving this conversation.

> | **credential_status_validity**| Numerical value indicating the validity of the Credential linked to the Status Assertion, describing its state, mode, condition or stage. The value MUST be from the IANA registry (TBD). Status validity parameter is REQUIRED, and the Verifier MUST verify its presence and value to assess the Credential's validity. | this specification |

Reviewer: ```suggestion
| **credential_status_validity**| Numerical value indicating the validity of the Credential linked to the Status Assertion, describing its state, mode, condition or stage. The value MUST be from the IANA registry ( as described in Section 7.1 of draft-ietf-oauth-status-list). Status validity parameter is REQUIRED, and the Verifier MUST verify its presence and value to assess the Credential's validity. | this specification |
```

Reviewer: This will need adjustment to link up properly as a well formed reference

Reviewer: I also cross filed this issue regarding the discussion on reserved for private use range here: https://github.com/oauth-wg/draft-ietf-oauth-status-list/issues/181

> | **credential_status_validity**| String value indicating the absolute validity of the Credential linked to the Status Assertion, describing its state, mode, condition or stage. Status validity MUST be `valid`, `invalid` or `suspended`. Status validity parameter is REQUIRED, and the Verifier MUST verify its presence and value to assess the Credential's validity. | this specification |

Reviewer: ```suggestion
| **credential_status_validity**| String value indicating the absolute validity of the Credential linked to the Status Assertion, describing its state, mode, condition or stage. The value MUST be from the IANA registry (TBD). Status validity parameter is REQUIRED, and the Verifier MUST verify its presence and value to assess the Credential's validity. | this specification |
```

the Wallet Instance to hash the Digital Credential for which the
Status Assertion is requested,  using one of the hash algorithms listed
in the [IANA-HASH-REG].
- `credential_status_detail_supported`. OPTIONAL. JSON array that outlines the details of each Credential's validity status supported by the Credential Issuer. This parameter MAY be used to extend the values defined in Section [Status Assertion](#status-assertion). Each entry MUST contain the following parameters:

>     - `credential_status_validity`. Numerical value indicating the validity of the Credential.  

Reviewer 3: ```suggestion
    - `credential_status_validity`. Numerical value indicating the validity of the Credential.
```

    - `state`. String value of a Credential status supported.

>     - `description`. String containing the human-readable description of the status related to this object.

Reviewer 2: @marinaado , I added this additional optional parameter in the metadata of a CI. The idea is that a CI MAY use this parameter to extend the values defined within this specification (0, 1, 2) and the Wallet can obtain from the CI's Metadata the detailed info to prompt the User.  
WDYT about this? 

@OR13 , @peppelinux  ^^

Author: Agreed that the IANA registry reserved range for private use should be used to communicate values outside of the ones defined in the IANA registry.

# Detailed Status Assertions

> Status Assertions can introduce a more accurate level of detail about the credential status.

Reviewer 3: ```suggestion
Status Assertions can introduce a more accurate level of detail about the Credential status.
```

# Detailed Status Assertions

Status Assertions can introduce a more accurate level of detail about the Credential status.
This enables Verifier policies to be conditioned on the presence of authorative information.
This section proposes syntax to support detailed assertions.
The `credential_status_validity` claim MUST be present.
The `credential_status_detail` claim MAY be present and if present MUST be an object.
The semantics of the claims within the `credential_status_detail` object are determined by the Credential Issuer.

> An example of bolean detail status is:

Reviewer 2: ```suggestion
An example of boolean detail status is:
```

    "exp": 1504785536,
    "credential_hash": "xnlAq6Ma8fgu1z4hdGphJnKLulaVHpLCFeZFUGpQ2dA",
    "credential_hash_alg": "sha-256",

>     "credential_status_validity": suspended,

Reviewer: ```suggestion
    "credential_status_validity": "suspended",
```

>     "credential_status_validity": suspended,

Reviewer 3: ```suggestion
    "credential_status_validity": "suspended",
```

    "exp": 1504785536,
    "credential_hash": "xnlAq6Ma8fgu1z4hdGphJnKLulaVHpLCFeZFUGpQ2dA",
    "credential_hash_alg": "sha-256",
    "credential_status_validity": 3,
    "credential_status_detail": {

>       "revoked": false,
>       "suspended": true,

Reviewer 3: ```suggestion
      ...
```

    "exp": 1504785536,
    "credential_hash": "xnlAq6Ma8fgu1z4hdGphJnKLulaVHpLCFeZFUGpQ2dA",
    "credential_hash_alg": "sha-256",

>     "credential_status_validity": invalid,

Reviewer: ```suggestion
    "credential_status_validity": "invalid",
```

>     "credential_status_validity": invalid,

Reviewer 3: ```suggestion
    "credential_status_validity": "invalid",
```

    "credential_status_detail": {

>       "state": "Document lost"

Reviewer 3: ```suggestion
      "state": "document lost"
```

    "exp": 1504785536,
    "credential_hash": "xnlAq6Ma8fgu1z4hdGphJnKLulaVHpLCFeZFUGpQ2dA",
    "credential_hash_alg": "sha-256",
    "credential_status_validity": 3,
    "credential_status_detail": {

>       "state": "suspended",

Reviewer 3: ```suggestion
      ...
```

    "exp": 1504785536,
    "credential_hash": "xnlAq6Ma8fgu1z4hdGphJnKLulaVHpLCFeZFUGpQ2dA",
    "credential_hash_alg": "sha-256",

>     "credential_status_validity": valid,

Reviewer: ```suggestion
    "credential_status_validity": "valid",
```

>     "credential_status_validity": valid,

Reviewer 3: ```suggestion
    "credential_status_validity": "valid",
```
